config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleMetadataBuilder.java

```diff
@@ -33,13 +34,15 @@ public BundleMetadata build(final AnnotatedBundle annotatedBundle, final Annotat
         final Encass encass = (Encass) annotatedEntity.getEntity();
         final String bundleName = annotatedBundle.getBundleName();
         final String name = bundleName.substring(0, bundleName.indexOf(projectVersion) - 1);
+        final boolean isPolicyOrEncassReusable = annotatedBundle.getEntities(Policy.class).entrySet().stream().anyMatch(entity -> entity.getValue().isReusable()) ||
+                annotatedBundle.getEntities(Encass.class).entrySet().stream().anyMatch(entity -> entity.getValue().isReusable());
 
         BundleMetadata.Builder builder = new BundleMetadata.Builder(encass.getType(), encass.getGuid(), name,
                 projectGroupName, projectVersion);
         builder.description(annotatedEntity.getDescription());
         builder.environmentEntities(getEnvironmentDependenciesMetadata(dependentEntities));
         builder.tags(annotatedEntity.getTags());
-        builder.reusableAndRedeployable(true, annotatedEntity.isRedeployable());
+        builder.reusableAndRedeployable(true, !isPolicyOrEncassReusable || annotatedEntity.isRedeployable());
```

Comment:
Please add JUnits to cover these 2 cases in metadata.

Reply:
Done